src/screens/SendSynthetic/SendSyntheticAmount.js

```diff
@@ -0,0 +1,382 @@
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import * as React from 'react';
+import { Keyboard, TouchableOpacity } from 'react-native';
+import { NavigationScreenProp } from 'react-navigation';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { SDK_PROVIDER } from 'react-native-dotenv';
+import styled from 'styled-components/native';
+import t from 'tcomb-form-native';
+import get from 'lodash.get';
+import isEmpty from 'lodash.isempty';
+import debounce from 'lodash.debounce';
+
+// components
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import Button from 'components/Button';
+import Spinner from 'components/Spinner';
+import { BaseText, Label, TextLink } from 'components/Typography';
+import { Wrapper } from 'components/Layout';
+
+// actions
+import { initSyntheticsServiceAction } from 'actions/syntheticsActions';
+
+// utils, services
+import { fontStyles, spacing, UIColors } from 'utils/variables';
+import { formatAmount, formatFiat, isValidNumber, parseNumber } from 'utils/common';
+import { getAssetData, getAssetsAsList, getRate } from 'utils/assets';
+import syntheticsService from 'services/synthetics';
+import { getAmountFormFields } from 'utils/formHelpers';
+
+// constants
+import { defaultFiatCurrency, PLR } from 'constants/assetsConstants';
+import { SEND_SYNTHETIC_CONFIRM, SEND_TOKEN_CONFIRM } from 'constants/navigationConstants';
+
+// selectors
+import { accountAssetsSelector } from 'selectors/assets';
+
+// models
+import type { Asset, Assets, Rates } from 'models/Asset';
+import type { SyntheticTransaction, TokenTransactionPayload } from 'models/Transaction';
+
+type Props = {
+  accountAssets: Assets,
+  supportedAssets: Asset[],
+  initSyntheticsService: Function,
+  navigation: NavigationScreenProp<*>,
+  rates: Rates,
+  baseFiatCurrency: string,
+  isOnline: boolean,
+};
+
+type State = {
+  value: ?{ amount: ?string },
+  submitPressed: boolean,
+  intentError: ?string,
+  inputHasError: boolean,
+};
+
+const { Form } = t.form;
+
+const generateFormStructure = (intentError: ?string, maxAmount: number, decimals: number) => {
+  const Amount = t.refinement(t.String, (amount): boolean => {
+    if (!isValidNumber(amount.toString()) || !!intentError) return false;
```

Comment:
I don't think enforcing boolean values is necessary on `if` conditions, we can avoid that extra noise.

```suggestion
    if (!isValidNumber(amount.toString()) || intentError) return false;
```